### PR TITLE
Set messages

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesMethod.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        SetMessagesResponse.Builder builder = SetMessagesResponse.builder();
+        processDestroy(request.getDestroy(), mailboxSession, builder);
+        return builder.build();
+    }
+
+    private void processDestroy(List<MessageId> destroy, MailboxSession mailboxSession, SetMessagesResponse.Builder builder) throws MailboxException {
+        MessageMapper<Id> messageMapper = mailboxSessionMapperFactory.createMessageMapper(mailboxSession);
+        destroy.stream()
+            .map(messageId -> delete(messageId, messageMapper, mailboxSession, builder))
+            .collect(Collectors.toList());
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesMethod.java]: you don't need to collect, you need foearch

Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesMethod.java]: Right

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetMessagesResponse.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+        private String accountId;
+        private String oldState;
+        private String newState;
+        private ImmutableList.Builder<Message> created;
+        private ImmutableList.Builder<MessageId> updated;
+        private ImmutableList.Builder<MessageId> destroyed;
+        private ImmutableList.Builder<SetError> notCreated;
+        private ImmutableList.Builder<SetError> notUpdated;
+        private ImmutableList.Builder<SetError> notDestroyed;
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetMessagesResponse.java]: It's actually a map

Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetMessagesResponse.java]: Right

**Comment by aduprat** [comment]
Replaced by #117

**Pull request closed without merging**